Libplanet/Blockchain/BlockChain.cs

```diff
@@ -472,6 +466,25 @@ internal BlockChain<T> Fork(HashDigest<SHA256> point)
                         break;
                     }
                 }
+
+                Block<T> pointBlock = Blocks[point];
+
+                var toUpdateAddresses = new HashSet<Address>();
+
+                for (var i = pointBlock.Index + 1; i <= Tip.Index; i++)
```

Comment:
(minor) We can omit block index lookup by looping reverse order using `Tip.PreviousHash` and `Blocks[Hash]`.